controllers/player.controller.js

```diff
@@ -1,43 +1,64 @@
 const Player = require('../models/player.model')
+const logs = require('../helpers/logsHelper')
 
-exports.hello = (req, res) => {
-  res.send('Hello world')
+exports.player_create = async (req, res) => {
+  try {
+    const player = await Player.findOneAndUpdate(
+      { name: req.body.name },
+      { $set: { name: req.body.name, score: req.body.score } },
+      { upsert: true }
+    )
+    res.status(200).send(player)
+  } catch (error) {
+    const requestId = req.get('kinto-request-id')
+    logs.logError(requestId, error)
+    res.status(400).send({ error: `player not saved: ${error}` })
+  }
 }
 
-exports.player_create = (req, res) => {
-  let player = new Player({
-    name: req.body.name,
-    score: req.body.score
-  })
-
-  player.save(function(err) {
-    if (err) {
-      return next(err)
-    }
-    res.send('Player Created successfully')
-  })
+exports.player_details = async (req, res) => {
+  try {
+    const player = await Player.findOne({ name: req.params.username })
```

Comment:
here you are assuming player is not gonna be found if `findOne` throws, can you confirm, because from what i know if `findOne` didn't find anything it will return null, and u'll need to check that here not in the catch